src/aspects/ec2-automated-shutdown/index.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { join } from 'path';
+import { Stack, Duration, IAspect } from 'aws-cdk-lib';
+import { Metric, Alarm, ComparisonOperator } from 'aws-cdk-lib/aws-cloudwatch';
+import { LambdaAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
+import { Instance, CfnInstance } from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement, Effect, AnyPrincipal } from 'aws-cdk-lib/aws-iam';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { Code, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { SnsEventSource } from 'aws-cdk-lib/aws-lambda-event-sources';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { IConstruct } from 'constructs';
+import { SecureFunction } from '../../constructs/secure-function';
+import { LambdaConfiguration } from '../../interfaces/lambda-configuration';
+
+/**
+ * Cleanup:
+ * EC2 -> Ec2 : done
+ * include name of instance in logical identifiers: done
+ * PascalCase for naming + identifiers:
+ */
+
+export interface CustomMetricConfig {
+    namespace: string;
+    metricName: string;
```

Comment:
We should create a new type under a `types` folder for this that can be retrieved from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/viewing_metrics_with_cloudwatch.html

For example

```
export type Ec2MetricName =
  | 'CPUUtilization'
  | 'DiskReadOps'
  | 'DiskWriteOps'
  | 'DiskReadBytes'
  | 'DiskWriteBytes'
  | 'NetworkIn'
  | 'NetworkOut'
  | 'NetworkPacketsIn'
  | 'NetworkPacketsOut'
  | 'StatusCheckFailed'
  | 'StatusCheckFailed_Instance'
  | 'StatusCheckFailed_System'
  | 'MetadataNoToken'
  | 'EBSReadOps'
  | 'EBSWriteOps'
  | 'EBSReadBytes'
  | 'EBSWriteBytes'
  | 'EBSIOBalance%'
  | 'EBSByteBalance%';
```

That way, users will automatically get this type hinting for all EC2 metrics available. (Double check that these all exist, this was autogenerated) ^